nautobot/docs/models/dcim/deviceredundancygroup.md

```diff
@@ -0,0 +1,359 @@
+                                    }
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+```
+
+### Retrieving the data - Secondary Failover Unit ("nyc-fw-secondary")
+
+An example data returned from Nautobot is presented below.
+
+```python
+>>> hostname_secondary = "nyc-fw-secondary"
+>>> secondary_unit_details = get_gql_failover_details(hostname_secondary).json
+```
+
+```json
+{
+    "data": {
+        "devices": [
+            {
+                "name": "nyc-fw-secondary",
+                "device_redundancy_group": {
+                    "name": "nyc-firewalls",
+                    "members": [
+                        {
+                            "name": "nyc-fw-primary",
+                            "device_redundancy_group_priority": 100,
+                            "interfaces": [
+                                {
+                                    "type": "VIRTUAL",
+                                    "name": "failover-link",
+                                    "ip_addresses": [
+                                        {
+                                            "host": "172.27.48.0",
+                                            "prefix_length": 31
+                                        }
+                                    ],
+                                    "parent_interface": {
+                                        "name": "gigabitethernet0/3",
+                                        "type": "A_1000BASE_T"
+                                    }
+                                }
+                            ]
+                        },
+                        {
+                            "name": "nyc-fw-secondary",
+                            "device_redundancy_group_priority": 50,
+                            "interfaces": [
+                                {
+                                    "type": "VIRTUAL",
+                                    "name": "failover-link",
+                                    "ip_addresses": [
+                                        {
+                                            "host": "172.27.48.1",
+                                            "prefix_length": 31
+                                        }
+                                    ],
+                                    "parent_interface": {
+                                        "name": "gigabitethernet0/3",
+                                        "type": "A_1000BASE_T"
+                                    }
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        ]
+    }
+}
+```
+
+### Processing the data - Primary Failover Unit ("nyc-fw-primary")
+
+Before rendering the data, we will process the data to simplify data variables within the template:
```

Comment:
I don't know this is advantageous for learning purposes. To me we should adjust the template (as would be done in Golden Config and make it easier to use an online jinja processing engine.

Reply:
This has been written in order to simplify the content of the template - I could store everything inside the template making the template quite un-readable.
Consider multiple nodes logic inside jinja2 template - it requires to get all members, identify local & remote nodes, get pair interfaces from remote nodes / local nodes etc.

Reply:
To me the value is being able to directly use it, especially in golden config. I’ll let other chime in though

Reply:
Fixed